cmd/build/v2/services.go

```diff
@@ -63,8 +56,15 @@ func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model
 		return nil, nil
 	}
 
+	svcToDeployMap := map[string]bool{}
+	for _, svc := range svcToDeploy {
+		svcToDeployMap[svc] = true
+	}
 	svcsToBuildList := []string{}
 	for svc := range toBuild {
+		if _, ok := svcToDeployMap[svc]; len(svcToDeploy) > 0 && !ok {
+			continue
+		}
```

Comment:
@jLopezbarb Why are we doing this? I've traced [this issue](https://github.com/okteto/okteto/issues/2795) to this line and I'm not sure why is this here